src/transformers/trainer.py

```diff
@@ -1448,15 +1448,29 @@ def floating_point_ops(self, inputs: Dict[str, Union[torch.Tensor, Any]]):
             :obj:`int`: The number of floating-point operations.
         """
 
-        if isinstance(self.model, torch.nn.DataParallel) or isinstance(
-            self.model, torch.nn.parallel.DistributedDataParallel
-        ):
-            model = self.model.module
-        else:
-            model = self.model
+        model = self._actual_model(self.model)
 
         if hasattr(model, "floating_point_ops"):
             return model.floating_point_ops(inputs)
 
         else:
             return 0
+
+    @staticmethod
+    def _actual_model(
+        model: Union[torch.nn.DataParallel, torch.nn.parallel.DistributedDataParallel, torch.nn.modules.Module]
```

Comment:
nit: aren't them all subclasses of `torch.nn.Module`?

Reply:
I wanted to be explicit, but I can change it if you think that's required.